packages/editor-ui/src/composables/useRunWorkflow.ts

```diff
@@ -385,10 +384,10 @@ export function useRunWorkflow(useRunWorkflowOpts: { router: ReturnType<typeof u
 
 					if (
 						execution.finished ||
-						(previousExecutionId && execution.id === previousExecutionId) ||
 						['error', 'canceled', 'crashed', 'success'].includes(execution.status)
 					) {
 						workflowsStore.setWorkflowExecutionData(execution);
+						workflowsStore.activeExecutionId = null;
```

Comment:
since this is what was causing the race condition by resetting `activeExecutionId` before the push handler was done processing the `executionFinished` event, how is this PR addressing the race condition now?